### PR TITLE
feat: add missing function to export ed25519 seed

### PR DESCRIPTION
**Pull request by metalurgical**

### CODE REVIEW

File: demo/redirect-flow-example/src/App.tsx [hunk 1]
```diff
@@ -568,7 +568,11 @@ function App() {
         </button>
 
         <button onClick={async () => uiConsole(await coreKitInstance._UNSAFE_exportTssKey())} className="card">
-          [CAUTION] Export TSS Private Key
+          [CAUTION] Export TSS Private Key for SECP256k1
+        </button>
+
+        <button onClick={async () => uiConsole(await coreKitInstance._UNSAFE_exportTssEd25519Seed())} className="card">
+          [CAUTION] Export TSS Seed for ED25519
```
Comment (matthiasgeihs) [inline demo/redirect-flow-example/src/App.tsx:575]: The demo only ever supports one key type. No need to create two buttons. You could alter the button text and functionality based on `tssLib.keyType`?

File: src/interfaces.ts [hunk 2]
```diff
@@ -266,9 +266,14 @@ export interface ICoreKit {
   commitChanges(): Promise<void>;
 
   /**
-   * Export the user's current TSS MPC account as a private key
+   * Export the user's current SECP256k1 TSS MPC account as a private key
```
Comment (matthiasgeihs) [inline src/interfaces.ts:269]: This also works for ed25519. In which case it will output the private key scalar.

File: src/mpcCoreKit.ts [hunk 3]
```diff
@@ -784,11 +784,12 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
   }
 
   /**
-   * WARNING: Use with caution. This will export the private key.
+   * WARNING: Use with caution. This will export the private key for secp256k1 KeyType.
    *
    * Exports the private signing key for the current account index.
    */
   public async _UNSAFE_exportTssKey(): Promise<string> {
+    if (!(this.keyType === KeyType.secp256k1)) throw new Error("wrong keytype to call this method");
```
Comment (matthiasgeihs) [inline src/mpcCoreKit.ts:792]: We could let this also output the private key scalar for keyType != secp256k1. I think @himanshuchawla009 previously talked about potential use cases for this. I'm undecided, both seems fine.

Comment (matthiasgeihs) [inline src/mpcCoreKit.ts:792]:
```suggestion
    if (this.keyType !== KeyType.secp256k1) throw new Error("wrong keytype to call this method");
```

File: src/mpcCoreKit.ts [hunk 4]
```diff
@@ -807,6 +808,25 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
     return tssKey.toString("hex", FIELD_ELEMENT_HEX_LEN);
   }
 
+  /**
+   * WARNING: Use with caution. This will export the seed  for ed25519 KeyType.
+   *
+   * Exports the seed for the current account index.
+   */
+  public async _UNSAFE_exportTssEd25519Seed(): Promise<Buffer> {
+    if (!(this.keyType === KeyType.ed25519)) throw new Error("wrong key type to call this method");
```
Comment (matthiasgeihs) [inline src/mpcCoreKit.ts:817]:
```suggestion
    if (this.keyType !== KeyType.ed25519) throw new Error("wrong key type to call this method");
```

File: src/mpcCoreKit.ts [hunk 5]
```diff
@@ -807,6 +808,25 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
...
+    if (!this.state.factorKey) throw new Error("factorKey not present");
+    if (!this.state.signatures) throw new Error("signatures not present");
+
+    const exportEd25519Seed = await this.tKey._UNSAFE_exportTssEd25519Seed({
+      factorKey: this.state.factorKey,
+      authSignatures: this.state.signatures,
+      selectedServers: [],
```
Comment (matthiasgeihs) [inline src/mpcCoreKit.ts:824]:
```suggestion
```

should work without selecting servers, no? i think having `selectedServers: []` is semantically unclear what it does.

**Comment by metalurgical** [comment]
Resolves #157

File: src/mpcCoreKit.ts [hunk 6]
```diff
@@ -807,6 +807,24 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
     return tssKey.toString("hex", FIELD_ELEMENT_HEX_LEN);
   }
 
+  /**
+   * WARNING: Use with caution. This will export the seed  for ed25519 KeyType.
+   *
+   * Exports the seed for the current account index.
```
Comment (matthiasgeihs) [inline src/mpcCoreKit.ts:813]:
```suggestion
```

Sorry, overlooked this. Account indexes are not supported currently for ed25519.

Comment (metalurgical) [reply src/mpcCoreKit.ts]: Removed that part of the comment.

**Comment by matthiasgeihs** [comment]
Revised the comment. Also clarified that export seed is only available for ed25519 accounts that were using import key flow.

**Comment by himanshuchawla009** [comment]
> Revised the function comments. Also clarified that export seed is only available for ed25519 accounts that were using import key flow.

does this means that accounts created with as default /without import wont be able to export seed?

**Comment by metalurgical** [comment]
> > Revised the function comments. Also clarified that export seed is only available for ed25519 accounts that were using import key flow.
> 
> does this means that accounts created with as default /without import wont be able to export seed?

I would have thought that the seed should always be available, to keep it implementation neutral. Appears not to be the case though.

**Comment by matthiasgeihs** [comment]
Yeah, this is because seed is not available when we just create account from shares. To have the seed, we need to start with generating the seed, which is our import seed flow.

Maybe we can further improve the documentation to clearly state that if the user wants the ability to export seed, it must generate seed on frontend and use import seed flow to initialize account.

**Comment by matthiasgeihs** [comment]
have extended documentation. i think a deeper explanation (about why ed25519 uses a seed and how that relates to the scalar) should be included in the docs.

**Pull request merged**